girder/models/model_base.py

```diff
@@ -1416,10 +1416,7 @@ def load(self, id, level=AccessType.ADMIN, user=None, objectId=True,
         :raises ValidationException: If an invalid ObjectId is passed.
         :returns: The matching document, or None if no match exists.
         """
-        # Warn of str type deprecation for `fields` param
         if isinstance(fields, six.string_types):
-            logger.warning('String data type for fields param is deprecated, \
```

Comment:
The underlying PyMongo's `projection` parameter only accepts a `list` or a `dict`. Personally, I think it encourages code sloppiness to have internal APIs that attempt to be overly permissive in what they'll tolerate, as it leads to needing defensive programming code and validations / coercions in internally-called methods. I'd rather internal APIs just have well-documented preconditions, so we can make stronger assumptions about variable types and parameter usage.

Note, these preconditions can still be abstract and yet rigorous. For example:
> parameter `x` is an `Iterable`

instead of 

> parameter `X` can be a `list` or a `tuple`; and if it's a `str`, we'll convert it to a `list` for you; and if it's a `dict`, we'll take its `.keys()` for you; and if it's some other iterable, we probably didn't consider that, so the behavior is undefined-ish...

Practically though, I think that tolerating `fields` as a `str` exists in a few different places, so if we're going to remove the coercion, it would be a bigger change than this one place.

Reply:
What's your conclusion? I don't have a preference either way.

Reply:
Can you un-un-deprecate this particular location, then let's address it more comprehensively and decide in another PR?